src/meta/processors/job/JobManager.cpp

```diff
@@ -587,6 +587,36 @@ bool JobManager::checkJobExist(GraphSpaceID spaceId,
   return false;
 }
 
+nebula::cpp2::ErrorCode JobManager::checkNotFinishedJobExist(GraphSpaceID spaceId,
+                                                             const cpp2::JobType& jobType) {
+  if (jobType == cpp2::JobType::DATA_BALANCE || jobType == cpp2::JobType::ZONE_BALANCE) {
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto jobPre = MetaKeyUtils::jobPrefix(spaceId);
+    auto retCode = kvStore_->prefix(kDefaultSpaceId, kDefaultPartId, jobPre, &iter);
+    if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
+      LOG(INFO) << "Fetch jobs failed, error: " << apache::thrift::util::enumNameSafe(retCode);
+      return retCode;
+    }
+
+    for (; iter->valid(); iter->next()) {
+      if (!MetaKeyUtils::isJobKey(iter->key())) {
+        continue;
+      }
+      auto tup = MetaKeyUtils::parseJobVal(iter->val());
+      auto type = std::get<0>(tup);
+      auto status = std::get<2>(tup);
+      if (type == cpp2::JobType::DATA_BALANCE || type == cpp2::JobType::ZONE_BALANCE) {
+        if (status != cpp2::JobStatus::FINISHED && status != cpp2::JobStatus::INVALID) {
+          LOG(INFO) << "There are some data balance or zone balance jobs that did not complete "
+                       "successfully";
+          return nebula::cpp2::ErrorCode::E_JOB_MAYBE_RECOVER;
```

Comment:
Do not understand "maybe recover". May change to "E_JOB_UNFINISHED"

Reply:
change to E_JOB_NEED_RECOVER